test/functorch/test_control_flow.py

```diff
@@ -1471,44 +1726,860 @@ def fct_pointwise(x, y):
+        inp = torch.randn(3, 10, 2, device=device)
+        expected_result = _fake_associative_scan(
+            add, inp, 1, reverse=reverse
+        ) @ torch.ones(2, 5, device=device)
+        result1 = fct_cmp(inp)
+        self.assertEqual(result1, expected_result)
+
+        # Chain with scan
+        def chain_fct_same_dim(inp):
+            o1 = associative_scan(
+                add, inp, 1, combine_mode=combine_mode, reverse=reverse
+            )
+            o2 = associative_scan(
+                add, o1, 1, combine_mode=combine_mode, reverse=reverse
+            )
+            return o2
+
+        torch.compiler.reset()
+        if compile_mode == "compile":
+            fct_cmp = torch.compile(chain_fct_same_dim, fullgraph=True, dynamic=False)
+        elif compile_mode == "compile_dynamic_shape":
+            fct_cmp = torch.compile(chain_fct_same_dim, fullgraph=True, dynamic=True)
+        elif compile_mode == "eager":
+            fct_cmp = torch.compile(chain_fct_same_dim, fullgraph=True, backend="eager")
+        else:
+            fct_cmp = chain_fct_same_dim
+
+        expected_result = _fake_associative_scan(
+            add,
+            _fake_associative_scan(add, inp, 1, reverse=reverse),
+            1,
+            reverse=reverse,
         )
-        result2 = associative_scan2(
-            fct_pointwise, inp, 0, combine_mode=combine_mode, reverse=reverse
+        result1 = fct_cmp(inp)
+        self.assertEqual(result1, expected_result)
+
+        # Chain with scan on different dim
+        def chain_fct_different_dim(inp):
+            o1 = associative_scan(
+                add, inp, 1, combine_mode=combine_mode, reverse=reverse
+            )
+            o2 = associative_scan(
+                add, o1, 0, combine_mode=combine_mode, reverse=reverse
+            )
+            return o2
+
+        torch.compiler.reset()
+        if compile_mode == "compile":
+            fct_cmp = torch.compile(
+                chain_fct_different_dim, fullgraph=True, dynamic=False
+            )
+        elif compile_mode == "compile_dynamic_shape":
+            fct_cmp = torch.compile(
+                chain_fct_different_dim, fullgraph=True, dynamic=True
+            )
+        elif compile_mode == "eager":
+            fct_cmp = torch.compile(
+                chain_fct_different_dim, fullgraph=True, backend="eager"
+            )
+        else:
+            fct_cmp = chain_fct_different_dim
+
+        expected_result = _fake_associative_scan(
+            add,
+            _fake_associative_scan(add, inp, 1, reverse=reverse),
+            0,
+            reverse=reverse,
         )
-        expected_result = _fake_associative_scan(fct_pointwise, inp, 0, reverse=reverse)
+        result1 = fct_cmp(inp)
         self.assertEqual(result1, expected_result)
-        self.assertEqual(result2, expected_result)
 
     @unittest.skipIf(not SM70OrLater, "triton")
```

Comment:
We don't need triton for scan tests, right? Please also look at all scan tests to make sure the skip is deleted.

Reply:
I have removed the skip triton for the `scan` operator tests.